metrics/segmentStats.py

```diff
@@ -0,0 +1,506 @@
+
+        Parameters
+        ----------
+        subGraphSkeleton: Networkx Graph
+            disjoint subgraph of the skeleton's networkx graph
+
+        cycle: list
+            list of nodes in the cycle
+
+        Returns
+        -------
+        subGraphSkeleton : Networkx graph
+            graph changed inplace with visited nodes removed
+
+        countDict: Dictionary
+            class instance variable changed in place with key as source node,
+            value is the number of segments attached to the source node
+
+        lengthDict: Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the length of the segment
+
+        tortuosityDict: Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the tortuosity of the segment (Nan but saved as zero)
+
+        contractionDictdict: Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the contraction of the segment (0) since it is a cycle
+
+        """
+        listOfPerms = list(itertools.permutations(self._branchPoints, 2))
+        for sourceOnTree, item in listOfPerms:
+            if nx.has_path(subGraphSkeleton, sourceOnTree, item) and sourceOnTree != item:
+                simplePaths = list(nx.all_simple_paths(subGraphSkeleton, source=sourceOnTree, target=item))
+                for simplePath in simplePaths:
+                    simplePath
+                    if sum([1 for point in simplePath if point in self._branchPoints]) == 2:
+                        if sourceOnTree not in self._visitedSources:
+                            self.countDict[sourceOnTree] = 1
+                            self._visitedSources.append(sourceOnTree)
+                        else:
+                            self.countDict[sourceOnTree] = self.countDict[sourceOnTree] + 1
+                        curveLength = self._getLengthAndRemoveTracedPath(subGraphSkeleton, simplePath)
+                        curveDisplacement = np.sqrt(np.sum((np.array(sourceOnTree) - np.array(item)) ** 2))
+                        self.lengthDict[self.countDict[sourceOnTree], sourceOnTree, item] = curveLength
+                        self.tortuosityDict[self.countDict[sourceOnTree], sourceOnTree, item] = curveLength / curveDisplacement
+                        self.contractionDict[self.countDict[sourceOnTree], sourceOnTree, item] = curveDisplacement / curveLength
+                        if curveDisplacement != 0.0:
+                            if np.log(curveDisplacement) != 0.0:
+                                self.hausdorffDimensionDict[self.countDict[sourceOnTree], sourceOnTree, item] = np.log(curveLength) / np.log(curveDisplacement)
+
+    def setStats(self):
```

Comment:
This method is impenetrable to a mere human mind such as my own. I get the feeling this is the main method that calls the other ones, but I'm not entirely sure.

I suggest:
a) Document what this method does, including each of the overall steps it takes. Of any of the methods in the class to document, this one might actually be the most important.
b) Split this up into smaller functions. 

Ideally, each method should operate at a single level of abstraction. That means that some functions (probably including this one) would only orchestrate calls to other methods without delving into the math themselves, while other functions would do the lower-level dirty work of manipulating sets and arrays and doing intricate numpy calls. That way, the orchestrating method can be the main point where people start to read the code, much like an abstract in a scientific paper. It should generally be one of the first methods defined, so that reading the code feels like reading an essay, with the first paragraph describing the structure of the rest of the document.